cli/azd/cmd/util_test.go

```diff
@@ -123,6 +125,17 @@ func Test_promptEnvironmentName(t *testing.T) {
 func Test_getSubscriptionOptions(t *testing.T) {
 	t.Run("no default config set", func(t *testing.T) {
 		mockContext := mocks.NewMockContext(context.Background())
+
+		ioc.Global.RegisterSingleton(func() azcli.AzCli {
+			return azcli.NewAzCli(mockContext.Credentials, azcli.NewAzCliArgs{
+				HttpClient: mockContext.HttpClient,
+			})
+		})
+		ioc.Global.RegisterSingleton(func() config.Manager {
+			return mockContext.ConfigManager
+		})
+		ioc.Global.RegisterSingleton(account.NewManager)
```

Comment:
Our unit tests shouldn't not really need to manually register any components.  In unit tests the component should be able to be easily instantiated to support test without the added complexity of dealing with IoC.